sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -73,16 +74,18 @@ case class InsertIntoHiveTable(
     query: LogicalPlan,
     overwrite: Boolean,
     ifPartitionNotExists: Boolean,
-    outputColumnNames: Seq[String]
+    outputColumnNames: Seq[String],
+    partitionColumns: Seq[Attribute],
+    bucketSpec: Option[BucketSpec],
+    options: Map[String, String],
+    fileFormat: FileFormat,
+    externalTmpPath: String,
+    @transient stagingDir: Path,
```

Comment:
what's the difference between `externalTmpPath` and `stagingDir`?

Reply:
for old hive version, externalTmpPath and stagingDir are the same.
https://github.com/apache/spark/blob/a3c837ae2eaf2c7ba08563b7afa0f96df8a4e80b/sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/SaveAsHiveFile.scala#L129-L136


for new hive version:
https://github.com/apache/spark/blob/a3c837ae2eaf2c7ba08563b7afa0f96df8a4e80b/sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/SaveAsHiveFile.scala#L189-L197

- externalTmpPath: `new Path(getExternalScratchDir(extURI, hadoopConf, stagingDir), "-ext-10000")`
- stagingDir: `getExternalScratchDir(extURI, hadoopConf, stagingDir)`